internal/fs/inode/dir_test.go

```diff
@@ -1228,6 +1380,7 @@ func (t *DirTest) CreateLocalChildFile_ShouldnotCreateObjectInGCS() {
 	result, err := t.in.LookUpChild(t.ctx, name)
 	AssertEq(nil, err)
 	AssertEq(nil, result)
+	ExpectEq(UnknownType, t.getTypeFromCache(name))
```

Comment:
@sethiay Ideally, should be RegularFileType. But apparently it hasn't been set up that way. It looks like a logic-gap in dir.go in

```
func (d *dirInode) LookUpChild(ctx context.Context, name string) (*Core, error)
```

I don't know if it's intentional (shouldn't be IMHO). 

LookUpChild first queries the type of `name` in its type-cache. When it doesn't find it, then it makes a GCS stat-call (through `lookUpFile`, and `lookUpExplicitDir`) to get its type. Both of those fail in this case, and thus a local-file's type in type-cache always remains UnknownType .

Reply:
Adding my understanding from my offline discussion with @ashmeenkaur ..

A local-file right now is not added to GCS until SyncFile. So, it is also not added to the type-cache until then, as type-cache is primarily meant to avoid getting the type information from GCS.

Reply:
The LookUpInode calls for local files don't reach LookUpChild method for local files hence the cache is not populated at that time. The only stat call is made during clobbered check ([in Attributes() method](https://github.com/GoogleCloudPlatform/gcsfuse/blob/a4cb9c59963d4db2c5eb7dabb4093006b1150047/internal/fs/inode/file.go#L402)) which is served from cache (populates the cache if entry is missing).

This in as per design in my opinion.

Reply:
thanks @ashmeenkaur I've captured in internal bug id 321892246 for this now. Marking this thread closed.